plugins/terminal.koplugin/main.lua

```diff
@@ -92,7 +92,36 @@ local Terminal = WidgetContainer:extend{
     terminal_data = ".",
 }
 
+function Terminal:isExecutable(file)
+    if os.execute(string.format("test -x %s", file)) == 0 then -- full path
+        return true
+    elseif os.execute(string.format("which %s 2>/dev/null 1>/dev/null", file)) == 0 then
+        return true
+    end
+end
+
+-- Try SHELL environment variable and some standard shells
+function Terminal:determineShellExecutable()
```

Comment:
While it does describe what it does, perhaps `getDefaultShellExecutable` would be more idiomatic? But as far as I'm concerned whatever you prefer.